### PR TITLE
virtio_fs: new case to test socket-group

### PR DESCRIPTION
**Pull request by xiagao**
Add new case to test --socket-group.
ID: 2016893
Signed-off-by: Xiaoling Gao <xiagao@redhat.com>

### CODE REVIEW

**Comment by xiagao** [comment]
(1/2) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.virtio_fs_socket_group.q35: PASS (105.76 s)
 (2/2) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_socket_group.q35: PASS (145.80 s)

**Comment by xiagao** [comment]
@YongxueHong @zhenyzha Could you help review? Thanks.

**Comment by zhenyzha** [comment]
Sorry, I have a confusion, why don't we directly add a variant to https://github.com/autotest/tp-qemu/blob/master/qemu/tests/cfg/virtio_fs_share_data.cfg#L69

like this:
```
    only default.default.with_cache.auto.default
    socket_group_test = yes
    socket_group = viofs_g
    fs_binary_extra_options = " --socket-group=${socket_group}"
    pre_command = groupadd -f ${socket_group}
    post_command = groupdel ${socket_group}
    cmd_get_sock = "ls -l %s"
```

If restrictions are added, will this affect the existing test loop?

**Comment by xiagao** [comment]
> Sorry, I have a confusion, why don't we directly add a variant to https://github.com/autotest/tp-qemu/blob/master/qemu/tests/cfg/virtio_fs_share_data.cfg#L69
> 
> like this:
> 
> ```
>     only default.default.with_cache.auto.default
>     socket_group_test = yes
>     socket_group = viofs_g
>     fs_binary_extra_options = " --socket-group=${socket_group}"
If we add it here, fs_binary_extra_options should be:
fs_binary_extra_options += " --socket-group=${socket_group}"

it will make the virtiofsd cmd be like below which isn't correct.
-o source=/home/test, --socket-group=xxx,cache=auto

If we want to create a right virtiofsd, we must move it back to '- with_cache', but that will lead to the cases name change.

>     pre_command = groupadd -f ${socket_group}
>     post_command = groupdel ${socket_group}
>     cmd_get_sock = "ls -l %s"
> ```
> 
> If restrictions are added, will this affect the existing test loop?

**Comment by zhenyzha** [comment]
I conducted a simple test, and the following configuration seems to meet the demand.

```
        - with_cache:
            variants:
                - auto:
                    fs_binary_extra_options += ",cache=auto"
                - always:
                    fs_binary_extra_options += ",cache=always"
                - none:
                    fs_binary_extra_options += ",cache=none"
        - @socket_group:
            only socket_group.default.default
            fs_binary_extra_options += " --socket-group=viofs_g"
```
The final generated daemon meets expectations:
`/usr/libexec/virtiofsd --socket-path=/tmp/avocado_x0rzl62c/avocado-vt-vm1-fs-virtiofsd.sock -o source=/root/avocado/data/avocado-vt/virtio_fs_test/ --socket-group=viofs_g`

Only one case is generated, and it has no effect on the loop.
If I am missing something, please feel free to share your opinion.
 (29/69) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.virtio_fs_share_data.arm64-pci: CANCEL: Test cancelled due to --dry-run (0.00 s)

**Comment by xiagao** [comment]
> I conducted a simple test, and the following configuration seems to meet the demand.
> 
> ```
>         - with_cache:
>             variants:
>                 - auto:
>                     fs_binary_extra_options += ",cache=auto"
>                 - always:
>                     fs_binary_extra_options += ",cache=always"
>                 - none:
>                     fs_binary_extra_options += ",cache=none"
>         - @socket_group:
>             only socket_group.default.default
>             fs_binary_extra_options += " --socket-group=viofs_g"
> ```
> 
> The final generated daemon meets expectations: `/usr/libexec/virtiofsd --socket-path=/tmp/avocado_x0rzl62c/avocado-vt-vm1-fs-virtiofsd.sock -o source=/root/avocado/data/avocado-vt/virtio_fs_test/ --socket-group=viofs_g`
> 
> Only one case is generated, and it has no effect on the loop. If I am missing something, please feel free to share your opinion. (29/69) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.virtio_fs_share_data.arm64-pci: CANCEL: Test cancelled due to --dry-run (0.00 s)

Yes, this way works well.thanks.

**Comment by xiagao** [comment]
(1/2) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.virtio_fs_share_data.socket_group.q35: PASS (120.62 s)
 (2/2) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_share_data.socket_group.q35: PASS (145.01 s)

**Comment by xiagao** [comment]
@zhenyzha Update, pls have a check,thanks.

**Comment by zhenyzha** [comment]
The result is as expected:

```
 (1/1) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.virtio_fs_share_data.socket_group.arm64-pci: PASS (97.20 s)
 (30/69) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.virtio_fs_share_data.socket_group.arm64-pci: CANCEL: Test cancelled due to --dry-run (0.00 s)
```

LGTM.Acked-by: Zhenyu Zhang <zhenyzha@redhat.com>

**Review by zhenyzha — APPROVED** [review]

File: qemu/tests/virtio_fs_share_data.py [hunk 1]
```diff
@@ -75,6 +75,22 @@ def get_stdev(file):
...
         return stdev
 
+    def check_socket_group():
+        """
+        check socket path's user group
+        """
+        cmd_get_sock = params["cmd_get_sock"]
+        for device in vm.devices:
+            if isinstance(device, qdevices.QVirtioFSDev):
+                sock_path = device.get_param("sock_path")
```
Comment (YongxueHong) [inline qemu/tests/virtio_fs_share_data.py:85]:
```suggestion
                sock_path = device.get_param("sock_path")
                break
```
We can reduce the loop times.

File: qemu/tests/virtio_fs_share_data.py [hunk 2]
```diff
@@ -75,6 +75,23 @@ def get_stdev(file):
...
+                break
+        logging.info(process.system_output(cmd_get_sock % sock_path))
+        sock_path_info = process.system_output(cmd_get_sock % sock_path)
```
Comment (YongxueHong) [inline qemu/tests/virtio_fs_share_data.py:88]: Why does execute `process.system_output(cmd_get_sock % sock_path) ` twice?

Comment (xiagao) [reply qemu/tests/virtio_fs_share_data.py]: Deleted it. I added it as debug info before.
Actually there is no need to keep.

**Review by YongxueHong — APPROVED** [review]

**Pull request merged**